styx-common/src/test/java/com/spotify/styx/storage/DatastoreStorageTest.java

```diff
@@ -182,97 +179,20 @@ public void shouldPersistNextScheduledRun() throws Exception {
   }
 
   @Test
-  public void shouldNotRemoveWorkflowWhenSettingDockerImage() throws Exception {
```

Comment:
I don't see a test where a Workflow is stored once, and then a second time with just a modified field (i.e. the Docker image), to then check that `storage.workflow(...)` does return the updated version.

Reply:
that is not a valid case. everything should be overwritten.